src/pathOrderOptimizer.cpp

```diff
@@ -160,15 +162,23 @@ int PathOrderOptimizer::getRandomPointInPolygon(int poly_idx)
     return rand() % polygons[poly_idx]->size();
 }
 
+static inline bool pointsAreCoincident(const Point& a, const Point& b)
+{
+    return vSize2(a - b) < 25; // points are closer than 5uM, consider them coincident
+}
+
 /**
 *
 */
-void LineOrderOptimizer::optimize()
+void LineOrderOptimizer::optimize(const Polygons* combing_boundary)
 {
     int gridSize = 5000; // the size of the cells in the hash grid. TODO
     SparsePointGridInclusive<unsigned int> line_bucket_grid(gridSize);
     bool picked[polygons.size()];
     memset(picked, false, sizeof(bool) * polygons.size());/// initialized as falses
+    loc_to_line = nullptr;
+    this->combing_boundary = combing_boundary;
```

Comment:
I think it makes more sense to set `combinb_boundary` in the constructor.

Reply:
OK, done that.